tests/countries/test_grenada.py

```diff
@@ -0,0 +1,347 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.countries.grenada import Grenada, GD, GRD
+from tests.common import CommonCountryTests
+
+
+class TestGrenada(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        years = range(1975, 2050)
+        super().setUpClass(Grenada, years=years)
+
+    def test_country_aliases(self):
+        self.assertAliases(Grenada, GD, GRD)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(Grenada(years=1974))
+
+    def test_special_holidays(self):
+        self.assertHoliday(
+            "2011-12-27",
+            "2013-02-22",
+            "2016-12-27",
+            "2018-03-14",
+            "2022-06-24",
+            "2022-12-27",
+            "2023-07-04",
+        )
+
+    def test_new_years_day(self):
+        name = "New Year's Day"
+        self.assertHolidayName(name, (f"{year}-01-01" for year in range(1975, 2050)))
+        dt = (
+            "1978-01-02",
+            "1984-01-02",
+            "1989-01-02",
+            "1995-01-02",
+            "2006-01-02",
+            "2012-01-02",
+            "2017-01-02",
+            "2023-01-02",
+            "2034-01-02",
```

Comment:
```suggestion
            "2006-01-02",
            "2012-01-02",
            "2017-01-02",
            "2023-01-02",
```
No need for early and future years, except in specific cases.

Reply:
This is a general approach that applies to _all_ holidays.

Reply:
For all observed holidays I have removed years before 2000 and kept the nearest future observed date. For National heroes day and Emancipation day(August 1), two dates in the nearest future have been kept because these holidays were added recently.